### PR TITLE
feat: Implement Task 6 - Repository Configuration Integration

### PR DESCRIPTION
**Pull request by mstriebeck**
Implements repository configuration integration for MCP Codebase Server as specified in US003-Single-Python-Repository-Discovery.md Task 6.

### New Module: codebase_repository_config.py
- **CodebaseRepositoryConfig** dataclass: Stores repository metadata
  - name, path, description, python_path fields
  - Path normalization and validation in __post_init__
- **CodebaseRepositoryConfigManager** class: Main configuration manager
  - Integrates with existing RepositoryManager
  - Filters repositories by language="python"
  - Validates repository paths and Python file existence
  - Provides get_python_repositories() for indexing workflows

### Key Features
- **Python Repository Filtering**: Identifies Python repos from repositories.json
- **Path Validation**: Ensures directories exist, are readable, and contain Python files
- **Python File Detection**: Recursively searches for .py and .pyi files
- **Error Handling**: Graceful handling of invalid repositories
- **Factory Function**: create_codebase_repository_config_manager() for easy instantiation

### Integration Points
- **Existing RepositoryManager**: Reuses established configuration patterns
- **repositories.json Format**: No breaking changes to existing configuration
- **Multi-language Support**: Filters Python repos while preserving others

### Unit Tests (28 tests)
- CodebaseRepositoryConfig dataclass validation
- Path normalization and error handling
- Python file detection logic (.py, .pyi, nested directories)
- Repository filtering and validation
- Error scenarios and edge cases

### Integration Tests (7 tests)
- Real repositories.json file parsing
- Git repository integration (required by existing RepositoryManager)
- Nested Python file detection
- Python stub file (.pyi) support
- Configuration validation workflows
- Factory function testing

### Test Coverage: 95%
- Comprehensive coverage of all major code paths
- Edge case handling and error scenarios
- Integration with existing codebase patterns

### Code Quality
- ✅ All ruff linting checks pass
- ✅ All mypy type checking passes
- ✅ Modern Python type annotations (| None vs Optional)
- ✅ Proper dependency injection patterns

### Documentation
- Comprehensive docstrings for all public methods
- Type hints for all parameters and return values
- Clear error messages and logging

### Validation
- Real-world testing with current github-agent repository
- Successfully identifies Python repositories
- Validates configuration correctly

## Task Requirements Met

✅ **Read existing repositories.json configuration** ✅ **Filter repositories with language="python"**
✅ **Validate repository paths exist and contain Python files** ✅ **Extract repository metadata (name, path, description)** ✅ **Support for python_path field for virtual environment detection** ✅ **Integrate with existing configuration validation** ✅ **Comprehensive unit and integration tests**
✅ **No breaking changes to existing functionality**

This implementation provides the foundation for repository-aware codebase indexing while maintaining compatibility with existing multi-repository GitHub MCP server infrastructure.

## Next Steps
Task 6 is complete. Ready to proceed with Task 7: Server Startup and Indexing Orchestration to automatically index configured Python repositories on server startup.

Tests: All 35 tests passing
Coverage: 95% on new module
Code Quality: All checks passing

Amp-Thread: https://ampcode.com/threads/T-863d0fab-8185-408b-a36a-537d5e8c7931

### CODE REVIEW

File: codebase_repository_config.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        """Get list of Python repositories configured for indexing.
+
+        Returns:
+            List of CodebaseRepositoryConfig objects for Python repositories
+        """
+        python_repos = []
+
+        for repo_name, repo_config in self.repository_manager.repositories.items():
+            # Filter for Python repositories
+            if repo_config.language == "python":
```
Comment (mstriebeck) [inline codebase_repository_config.py:60]: actually, we should make language an enum with only python and swift as its values. when we create the repo_config, we translate the string to the enum. That way we can make enum comparisons everywhere which will be much more robust than strings.

File: tests/test_codebase_repository_config.py [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+        assert config.python_path is None
+
+
+class TestCodebaseRepositoryConfigManager:
+    """Test the CodebaseRepositoryConfigManager class."""
+
+    def test_initialization(self):
+        """Test manager initialization."""
+        mock_repo_manager = Mock(spec=RepositoryManager)
```
Comment (mstriebeck) [inline tests/test_codebase_repository_config.py:87]: We have a MockRepositoryManager that we should pass into this test and the next ones via its fixture

File: tests/test_codebase_repository_config.py [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+        mock_repo_manager.repositories = {"swift-repo": mock_repo_config}
+
+        manager = CodebaseRepositoryConfigManager(mock_repo_manager)
+        python_repos = manager.get_python_repositories()
+
+        assert python_repos == []
+
+    @patch(
+        "codebase_repository_config.CodebaseRepositoryConfigManager._validate_python_repository"
```
Comment (mstriebeck) [inline tests/test_codebase_repository_config.py:117]: create and use a proper mock object here - not a patch

File: codebase_repository_config.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+import logging
+import os
+from dataclasses import dataclass
+
+from constants import Language
+from repository_manager import RepositoryManager
+
+
+@dataclass
+class CodebaseRepositoryConfig:
```
Comment (mstriebeck) [inline codebase_repository_config.py:20]: why did we create the CodebaseRepositoryConfig? We could just use the RepositoryConfig.

File: codebase_repository_config.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+        Returns:
+            List of CodebaseRepositoryConfig objects for Python repositories
+        """
+        python_repos = []
+
+        for repo_name, repo_config in self.repository_manager.repositories.items():
+            # Filter for Python repositories
+            try:
+                repo_language = Language(repo_config.language)
```
Comment (mstriebeck) [inline codebase_repository_config.py:62]: No, we need to replace language in RepositoryConfig with the enum!

File: mcp_master.py [hunk 6]
```diff
@@ -123,7 +123,7 @@ def description(self) -> str:
 
     @property
     def language(self) -> str:
-        return self.repository_config.language
+        return self.repository_config.language.value
```
Comment (mstriebeck) [inline mcp_master.py:126]: No, we don't return the value but the language. Everywhere internally, we are using the enum - not the string.

File: mcp_worker.py [hunk 7]
```diff
@@ -98,7 +98,7 @@ def __init__(self, repository_config: RepositoryConfig):
         self.repo_path = repository_config.path
         self.port = repository_config.port
         self.description = repository_config.description
-        self.language = repository_config.language
+        self.language = repository_config.language.value
```
Comment (mstriebeck) [inline mcp_worker.py:101]: self.language needs to be of type Language
